rfc/update-api.md

```diff
@@ -0,0 +1,105 @@
+# Update API
+
+The update API should expose easy, straightforward, extensible and forward compatible version information to the Constellation CLI (and possibly more consumers).
+
+Design goals:
+
+- Simple
+- Can be implemented using static HTTP file server
+- Generic over different kinds of resources
+    - OS image versions
+    - Microservice versions
+    - CLI versions
+    - Kubernetes versions
+
+The folowing HTTP endpoints are available:
+
+- `GET /constellation/v1/updates/latest/` contains files showing the latest version available
+    - `image.json` contains the latest image version
+    - `microservice.json` contains the latest microservice version
+    - `cli.json` contains the latest cli version
+    - `kubernetes.json` contains the latest supported version of Kubernetes
+- `GET /constellation/v1/updates/major/<major-version>/` contains files with version information for this major version
+    - `image.json` contains a list of all minor image versions that belong to a major version
+    - `microservice.json` contains a list of all minor microservice versions that belong to a major version
+    - `cli.json` contains a list of all minor cli versions that belong to a major version
+    - `kubernetes.json` contains a list of all supported minor version of Kubernetes that belong to a major version
+- `GET /constellation/v1/updates/minor/<minor-version>/` contains files with version information for this minor version
+    - `image.json` contains a list of all patch image versions that belong to a minor version
+    - `microservice.json` contains a list of all patch microservice versions that belong to a minor version
+    - `cli.json` contains a list of all patch cli versions that belong to a minor version
+    - `kubernetes.json` contains a list of all supported patch version of Kubernetes that belong to a minor version
```

Comment:
Should we combine some of those? I.e. I can only obtain a new microservice version through updating the CLI. So it is more of a meta information about the specific CLI version. I think we should at least somehow reflect which microservice version and Kubernetes version a specific CLI supports.

Reply:
I think we should not impose version constraints early on in this API. Keeping them separate allows us to change compatibility checks between resources more easily in the CLI.